crates/miden-lib/src/note/mod.rs

```diff
@@ -83,6 +83,44 @@ pub fn create_p2idr_note<R: FeltRng>(
     Ok(Note::new(vault, metadata, recipient))
 }
 
+/// Generates a hybrid P2ID note - pay to id with optional recall after a certain block height and
+/// optional timelock.
```

Comment:
nit: should probably say "Generates an extended P2ID note - pay-to-ID with optional..."